include/simdjson/generic/numberparsing.h

```diff
@@ -651,18 +654,21 @@ simdjson_unused simdjson_really_inline simdjson_result<int64_t> parse_integer(co
 
   // If there were no digits, or if the integer starts with 0 and has more than one digit, it's an error.
   int digit_count = int(p - start_digits);
-  if (digit_count == 0 || ('0' == *start_digits && digit_count > 1)) { return NUMBER_ERROR; }
-  if (!jsoncharutils::is_structural_or_whitespace(*p)) { return NUMBER_ERROR; }
+  if (digit_count == 0) { return INCORRECT_TYPE; }
+  if ('0' == *start_digits && digit_count > 1) { return NUMBER_ERROR; }
+  if (!jsoncharutils::is_structural_or_whitespace(*p)) {
+    return (*p == '.' || *p == 'e' || *p == 'E') ? INCORRECT_TYPE : NUMBER_ERROR;
+  }
```

Comment:
Same story... it might pay to gave a dedicated table that returns the error_code: we need or success. We do a lookup anyhow, we might as well get the error code for free? Fewer branches and instructions at the cost of a new small table.